File: examples/typescript/build_outputs.mjs
import { spawnSync } from "node:child_process";
import { dirname, resolve } from "node:path";
import { fileURLToPath } from "node:url";

function main() {
  const appRoot = dirname(fileURLToPath(import.meta.url));
  const repoRoot = resolve(appRoot, "..", "..");
  const command = generatorCommand();
  const result = spawnSync(
    command[0],
    [...command.slice(1), ...process.argv.slice(2)],
    {
      cwd: repoRoot,
      stdio: "inherit",
    },
  );

  if (result.status !== 0) {
    process.exit(result.status ?? 1);
  }
}

function generatorCommand() {
  if (process.env.NEXUS_API_GEN_BIN) {
    return [process.env.NEXUS_API_GEN_BIN, "build-examples", "--lang", "typescript"];
  }

  return ["cargo", "build-examples", "--lang", "typescript"];
}

main();

File: advanced/samples/python/build_outputs.py
from __future__ import annotations

import os
from pathlib import Path
import subprocess
import sys


def main() -> None:
    app_root = Path(__file__).resolve().parent
    repo_root = app_root.parent.parent
    _ = subprocess.run(
        generator_command() + ["--lang", "python", *sys.argv[1:]],
        check=True,
        cwd=repo_root,
    )


def generator_command() -> list[str]:
    if configured_binary := os.environ.get("NEXGEN_BIN"):
        return [configured_binary, "build-examples"]

    return ["cargo", "build-examples"]


if __name__ == "__main__":
    main()
